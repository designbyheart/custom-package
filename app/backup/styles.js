// @flow
import { StyleSheet, Dimensions, Platform } from 'react-native'
import {
  PADDING_HORIZONTAL,
  PADDING_VERTICAL,
  SHORT_DEVICE,
  color,
  grey,
} from '../common/styles/constant'

const { width, height } = Dimensions.get('window')
const SPACE_FILLER = 50
const amountSpacing = height > SHORT_DEVICE ? PADDING_VERTICAL * 2 : 0
const submitButtonHeight = height > SHORT_DEVICE ? 63 : 43
const showRecoveryButtonHeight = height > SHORT_DEVICE ? 150 : 100
const inputBoxHeight = height > SHORT_DEVICE && Platform.OS === 'ios' ? 137 : 40

const styles = StyleSheet.create({
  tabContainer: {
    maxHeight: 60,
  },
  backgroundImage: {
    flex: 1,
    position: 'absolute',
    transform: [{ rotate: '135deg' }],
  },
  headerCloseIcon: {
    marginRight: PADDING_HORIZONTAL,
    marginBottom: PADDING_VERTICAL,
    alignSelf: 'flex-end',
  },
  headerIcon: {
    marginRight: PADDING_HORIZONTAL,
    marginTop: 15,
    alignSelf: 'flex-end',
  },
  headerBackIcon: {
    marginLeft: PADDING_HORIZONTAL,
    marginTop: 15,
    alignSelf: 'flex-end',
  },
  headerSpacer: {
    height: '100%',
    width: SPACE_FILLER,
  },
  showRecoveryPassphrase: {
    width: '86%',
    height: showRecoveryButtonHeight,
    borderRadius: 5,
    alignSelf: 'center',
    backgroundColor: color.bg.tertiary.color,
    elevation: 7,
    marginBottom: 60,
  },
  genRecovery: {
    backgroundColor: color.bg.eleventh.color,
  },
  genRecoveryWrapper: {
    height: 80,
  },
  genRecoveryHeader: {
    width: '100%',
  },
  genRecoveryPhraseContainer: {
    position: 'absolute',
    top: 20,
    left: 20,
    right: 20,
  },
  genRecoveryPhrase: {
    textAlign: 'center',
    color: color.bg.eleventh.color,
    fontSize: 20,
    lineHeight: 30,
    fontFamily: 'Lato',
    fontWeight: '900',
  },
  genRecoveryText: {
    marginBottom: 20,
    alignSelf: 'center',
    fontSize: 22,
    fontWeight: '600',
    fontFamily: 'Lato',
  },
  genRecoveryMessage: {
    marginBottom: 40,
    paddingHorizontal: 20,
    alignSelf: 'center',
    fontSize: 18,
    lineHeight: 22,
    fontWeight: '500',
    fontFamily: 'Lato',
  },
  genRecoverySecondMessage: {
    marginBottom: 40,
    paddingHorizontal: 20,
    alignSelf: 'center',
    fontSize: 18,
    lineHeight: 22,
    fontWeight: 'bold',
    fontFamily: 'Lato',
  },
  genRecoverySmallMessage: {
    fontSize: 14,
    lineHeight: 17,
    fontWeight: 'bold',
    fontFamily: 'Lato',
    marginBottom: 30,
  },
  verifyMainContainer: {
    flex: 1,
    backgroundColor: color.bg.twelfth.color,
  },
  verifyMainText: {
    paddingHorizontal: 20,
    lineHeight: 22,
  },
  inputBox: {
    marginBottom: 24,
    marginRight: 20,
    marginLeft: 20,
    height: inputBoxHeight,
    borderRadius: 10,
    backgroundColor: 'rgba(0,0,0,0.33)',
    color: 'white',
    padding: 10,
    textAlignVertical: 'top',
  },
  submitButton: {
    marginLeft: '2%',
    marginRight: '2%',
    marginBottom: 15,
    borderRadius: 5,
    shadowColor: grey,
    shadowOffset: {
      width: 1,
      height: 1,
    },
    shadowRadius: 5,
    shadowOpacity: 0.3,
    elevation: 7,
    height: height > SHORT_DEVICE ? 63 : 43,
  },
  backgroundImageVerify: {
    flex: 1,
    position: 'absolute',
  },
  verifyTitle: {
    fontSize: 22,
    marginBottom: 20,
    width: '100%',
  },
  exportBackup: {
    backgroundColor: color.bg.thirteenth.color,
  },
  wrapper: {
    paddingHorizontal: PADDING_HORIZONTAL,
  },
  title: {
    fontSize: 22,
    marginBottom: 20,
    width: '100%',
  },
  exportBackupText: {
    marginBottom: 20,
    backgroundColor: color.bg.thirteenth.color,
    alignSelf: 'center',
    fontSize: 22,
  },
  exportBackupMessage: {
    backgroundColor: color.bg.thirteenth.color,
    alignSelf: 'center',
    fontSize: 18,
  },
  exportBackupSmallMessage: {
    backgroundColor: color.bg.thirteenth.color,
    alignSelf: 'center',
    fontSize: 14,
  },
  backupComplete: {
    backgroundColor: color.bg.fourteenth.color,
  },
  backupCompleteHeader: {
    width: '100%',
  },
  backupCompleteText: {
    marginBottom: 20,
    backgroundColor: color.bg.fourteenth.color,
    alignSelf: 'center',
    fontSize: 22,
  },
  backupCompleteMessage: {
    backgroundColor: color.bg.fourteenth.color,
    alignSelf: 'center',
    fontSize: 18,
  },
  backupCompleteSmallMessage: {
    backgroundColor: color.bg.fourteenth.color,
    alignSelf: 'center',
    fontSize: 14,
  },
  alignItemsCenter: {
    marginBottom: 6,
    marginLeft: '5%',
    marginRight: '5%',
  },
  verticalSpacing: {
    paddingTop: amountSpacing,
  },
  customButtonColor: {
    fontWeight: '600',
    fontSize: 18,
  },
})

export default styles
