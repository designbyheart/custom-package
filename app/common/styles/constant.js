// @flow
// TODO:KS Add support for themes as well
// Color name taken from http://www.htmlcsscolor.com/
const zircon = '#ebebea'
export const nightRider = '#333333'
export const grey = '#777777'
export const whiteSmoke = '#f0f0f0'
export const mantis = '#85bf43'
const mantisRGB = {
  r: 133,
  g: 191,
  b: 67,
}
export const white = '#ffffff'

const sirocco = '#636564'
const corduroy = '#3f4140'
const eclipse = '#3f3f3f'
const matterhorn = '#535353'
const blackPearl = '#242b2d'
const dimGray = '#686868'
const nobel = '#a0a0a0'
const venetianRed = '#d0021b'
export const veniceBlue = '#2A5270'
const dodgerBlue = '#0d8ffc'
const black = '#000000'
const seaBuckthorn = '#f79347'
const pumpkin = '#F68127'
const olivine = '#97C85F'

export const color = {
  actions: {
    primary: pumpkin,
    secondary: seaBuckthorn,
    tertiary: mantis,
    quaternary: olivine,
    fifth: white,
    dangerous: venetianRed,
    none: white,
    primaryRGB: mantisRGB,
    font: {
      primary: white,
      secondary: white,
      tertiary: white,
      quaternary: white,
      fifth: mantis,
      dangerous: white,
    },
  },
  bg: {
    primary: {
      color: nightRider,
      font: {
        primary: white,
        secondary: zircon,
      },
    },
    secondary: {
      color: grey,
      font: {
        primary: white,
        secondary: matterhorn,
        tertiary: whiteSmoke,
        quaternary: dodgerBlue,
      },
    },
    tertiary: {
      color: whiteSmoke,
      font: {
        primary: nightRider,
        secondary: dimGray,
        tertiary: grey,
      },
    },
    quaternary: {
      color: mantis,
      font: {
        primary: white,
      },
    },
    fifth: {
      color: white,
      font: {
        primary: grey,
        secondary: zircon,
        tertiary: eclipse,
      },
    },
    sixth: {
      color: nightRider,
      font: {
        primary: white,
      },
    },
  },
}

export const font = {
  size: {
    XS: 12,
    S: 15,
    M: 17,
    ML: 20,
  },
}

export const PADDING_HORIZONTAL = 15
export const PADDING_VERTICAL = 8
export const MARGIN_BOTTOM = 4
export const OFFSET_1X = 10
export const OFFSET_2X = 20
export const OFFSET_3X = 30
export const OFFSET_4X = 40
export const OFFSET_5X = 50
export const OFFSET_6X = 60
export const OFFSET_9X = 95
export const barStyleDark = 'dark-content'
export const barStyleLight = 'light-content'
export const SHADOW_RADIUS = 7

export const bubbleSize = {
  XS: 40,
  S: 60,
  M: 80,
  L: 100,
  XL: 120,
  XXL: 140,
}

export const PIN_CODE_BORDER_BOTTOM = 4
