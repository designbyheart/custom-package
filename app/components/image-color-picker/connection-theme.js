// @flow
import React, { Component } from 'react'
import { View } from 'react-native'
import { connect } from 'react-redux'

import empty from '../../common/empty'
import type { ConnectionThemeProps } from './type-color-picker'

export class ConnectionTheme extends Component<
  void,
  ConnectionThemeProps,
  any
> {
  render() {
    const { shade, activeConnectionColor, children, style } = this.props

    const styles = style || empty
    const colorShade = shade || '1.0'
    const connectionThemeColor =
      Object.values(activeConnectionColor).join(',') + ',' + colorShade

    let childrenWithProps = React.Children.map(children, child =>
      React.cloneElement(child, {
        customColor: {
          backgroundColor: `rgba(${connectionThemeColor})`,
        },
      })
    )

    return (
      <View
        style={[styles, { backgroundColor: `rgba(${connectionThemeColor})` }]}
      >
        {childrenWithProps}
      </View>
    )
  }
}

const mapStateToProps = ({ connections }) => ({
  activeConnectionColor: connections.activeConnectionColor,
})

export default connect(mapStateToProps, null)(ConnectionTheme)
