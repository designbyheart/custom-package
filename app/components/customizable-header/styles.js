import { StyleSheet } from 'react-native'

import { color, font } from '../../common/styles/constant'

export const styles = StyleSheet.create({
  commonHeaderContainer: {
    position: 'absolute',
    width: '100%',
    backgroundColor: 'rgba(255, 255, 255, 0.8)',
  },
  primaryHeaderContainer: {
    height: 120,
    flexDirection: 'row',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 10,
  },
  primaryHeaderTextSection: {
    width: '50%',
    height: '100%',
  },
  primaryHeaderIconSection: {
    width: '50%',
    height: '100%',
    flexDirection: 'row-reverse',
  },
  primaryHeaderText: {
    fontFamily: 'Lato',
    fontWeight: 'bold',
    fontSize: 26,
    marginTop: 73,
    marginLeft: 16,
    color: '#505050',
  },
  primaryHeaderIcon: {
    marginTop: 74,
    marginRight: 6,
  },
  connectionDetailsHeaderContainer: {
    height: 136,
  },
  credentialDetailsHeaderContainer: {
    height: 175,
  },
})

export const primaryHeaderContainer = StyleSheet.flatten([
  styles.commonHeaderContainer,
  styles.primaryHeaderContainer,
])

export const connectionDetailsHeaderContainer = StyleSheet.flatten([
  styles.commonHeaderContainer,
  styles.connectionDetailsHeaderContainer,
])

export const credentialDetailsHeaderContainer = StyleSheet.flatten([
  styles.commonHeader,
  styles.credentialDetailsHeaderContainer,
])
