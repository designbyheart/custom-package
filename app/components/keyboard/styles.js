// @flow

import { StyleSheet, Dimensions } from 'react-native'
import { SHORT_DEVICE } from '../../common/styles/constant'
import { PUSH_NOTIFICATION_PERMISSION } from '../../push-notification/type-push-notification'

const { height } = Dimensions.get('window')

const ROW_MARGIN = height > SHORT_DEVICE ? 36 : 15
const FIRST_ROW_MARGIN = height > SHORT_DEVICE ? 24 : 8
const ROW_HEIGHT = height > SHORT_DEVICE ? 33 : 28

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 32, // TODO: update style guide to reflect new constant
  },
  firstRow: {
    height: ROW_HEIGHT,
    marginTop: FIRST_ROW_MARGIN,
  },
  row: {
    height: ROW_HEIGHT,
    marginTop: ROW_MARGIN,
  },
  number: {
    fontSize: 28, // TODO: update style guide to reflect new constant
    fontWeight: '500', // TODO: update style guide to reflect new constant
  },
})

export default styles
