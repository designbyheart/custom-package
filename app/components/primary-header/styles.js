// @flow
import { StyleSheet, Platform, PixelRatio } from 'react-native'

import { isBiggerThanVeryShortDevice } from '../../common/styles/constant'

const containerHeight = isBiggerThanVeryShortDevice ? 100 : 90

// original in icon{} and label{} marginTop was 73 and height was 120, thus the 47 to keep
// subtracting 47 from containerHeight keeps it relative to what is was.
const marginTop = containerHeight - 47

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    height: containerHeight,
    width: '100%',
    flexDirection: 'row',
    backgroundColor:
      Platform.OS === 'ios' ? 'rgba(255, 255, 255, 0.8)' : '#fff',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 10,
    elevation: Platform.OS === 'android' ? 8 : 0,
  },
  labelSection: {
    width: '75%',
    height: '100%',
  },
  iconsSection: {
    width: '25%',
    height: '100%',
    flexDirection: 'row-reverse',
  },
  label: {
    fontFamily: 'Lato',
    fontWeight: 'bold',
    fontSize: 26,
    marginTop,
    marginLeft: 16,
    color: '#505050',
  },
  icon: {
    marginTop,
    marginRight: 6,
  },
  blur: {
    position: 'absolute',
    left: 0,
    bottom: 0,
    width: '100%',
    height: 120,
  },
})

export { styles }
