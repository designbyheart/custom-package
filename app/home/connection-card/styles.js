// @flow
import { StyleSheet, Platform } from 'react-native'
import { color } from '../../common/styles/constant'

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 80,
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderBottomColor: color.border.primary,
  },
  newCardContainer: {
    backgroundColor: color.bg.sixteenth.color,
  },
  avatarSection: {
    height: '100%',
    width: '20%',
    paddingTop: 16,
    paddingLeft: 16,
  },
  infoSection: {
    width: '100%',
    height: '100%',
  },

  infoSectionTopRow: {
    flexDirection: 'row',
    width: '80%',
    height: '50%',
  },
  infoSectionBottomRow: {
    width: '80%',
    height: '50%',
  },
  companyNameSection: {
    width: '70%',
    height: '100%',
    justifyContent: 'flex-end',
  },
  descriptionSection: {
    width: '95%',
    height: '100%',
  },
  dateButtonSection: {
    width: '30%',
    height: '100%',
    flexDirection: 'row',
    alignItems: 'flex-end',
  },
  dateSection: {
    width: '70%',
    height: '100%',
    flexDirection: 'row',
    alignItems: 'flex-end',
    justifyContent: 'flex-end',
  },
  buttonSection: {
    height: '50%',
    width: '30%',
    justifyContent: 'flex-end',
  },
  companyNameText: {
    flex: 1,
    fontFamily: 'Lato',
    fontSize: 17,
    fontWeight: 'bold',
    color: color.textColor.darkgray,
  },
  descriptionText: {
    flex: 1,
    fontFamily: 'Lato',
    fontSize: 14,
    color: color.textColor.grey,
  },
  dateText: {
    fontFamily: 'Lato',
    fontSize: 11,
    color: color.textColor.mediumGray,
    marginBottom: 5,
  },
  newButtonSection: {
    height: '100%',
    width: '30%',
    alignItems: 'center',
    justifyContent: 'flex-end',
  },
  newLabel: {
    width: 64,
    height: 16,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: color.bg.twelfth.color,
    marginBottom: 5,
  },
  newLabelText: {
    fontFamily: 'Lato',
    fontSize: 11,
    fontWeight: '500',
    color: color.bg.primary.font.primary,
  },
})

export { styles }
