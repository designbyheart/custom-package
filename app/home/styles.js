// @flow
import { StyleSheet } from 'react-native'
import { measurements } from '../common/styles/measurements'
import { primaryHeaderHeight } from '../common/styles/constant'

const externalStyles = StyleSheet.create({
  container: {
    width: '100%',
    height: '100%',
    backgroundColor: '#fff',
  },
  flatListContainer: {
    width: '100%',
    height: '100%',
    backgroundColor: '#fff',
    paddingTop: primaryHeaderHeight,
  },
  flatListInnerContainer: {
    paddingBottom: 170,
  },
  blurContainer: {
    position: 'absolute',
    left: 0,
    bottom: 0,
    width: '100%',
    height: measurements.bottomBlurNavBarHeight,
  },
})

export { externalStyles }
