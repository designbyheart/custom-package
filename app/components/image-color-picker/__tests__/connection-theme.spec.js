import React from 'react'
import 'react-native'
import { ConnectionTheme } from '../connection-theme'
import renderer from 'react-test-renderer'
import { color } from '../../../common/styles/constant'

function props() {
  return { activeConnectionColor: color.actions.primaryRGB }
}

describe('<ConnectionTheme />', () => {
  it('should render default shade theme properly', () => {
    const connectionTheme = renderer
      .create(<ConnectionTheme {...props()} />)
      .toJSON()
    expect(connectionTheme).toMatchSnapshot()
  })
})
