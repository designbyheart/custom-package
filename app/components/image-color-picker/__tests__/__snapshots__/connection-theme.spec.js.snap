// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectionTheme /> should render default shade theme properly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "backgroundColor": "rgba(133,191,67,1.0)",
      },
    ]
  }
/>
`;
